### PR TITLE
Tus client

### PR DESCRIPTION
**Pull request by ruwang**
node tus client to upload files.

### CODE REVIEW

File: src/transferImaging.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+'use strict';
```
Comment (rssk) [inline src/transferImaging.js]: Does this just transfer imaging files? If so cool, if not I would recommend renaming this.

File: src/transferImaging.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * @param {opts} meta data from penny.json
+ * @return {Promis}
+ */
+module.exports = (opts) => {
+  const filePath = opts.filePath;
+  // Zip files for tus client upload
+  const zipFile = path.basename(filePath) + '.zip'; // eslint-disable-line
+  const output = fs.createWriteStream(zipFile);
+  output.on('close', () => {
+    tusClient(zipFile, uploadURL, (err) => {
```
Comment (rssk) [inline src/transferImaging.js]: Does the tusClient handle all dropout-reconnect issues? Or is there more to it?

File: src/tusNodeClient.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                req = request(optionsHead);
+                req.once('response', (res) => {
+                  if (res.statusCode === 200) {
+                    offset = res.headers.offset;
+                    if (offset) {
+                      logger.debug('offset:', res.headers.offset);
+                    }
+                    callback();
+                  } else {
+                    // always there are an exit :-)d
```
Comment (rssk) [inline src/tusNodeClient.js]: Not sure I understand this comment?

File: src/tusNodeClient.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+      request(options, (error, response) => {
+        // File created
+        if (!error && response.statusCode === 201) {
+          const location = response.headers.location;
+          let req;
+
+          // Status check (might not needed)
+          async.whilst(
+            () => { // eslint-disable-line
```
Comment (rssk) [inline src/tusNodeClient.js]: Why is this line disabled?

File: src/tusNodeClient.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+          const optionsSendFile = {
+            url: location,
+            method: 'PATCH',
+            headers: {
+              'Content-Type': 'application/offset+octet-stream',
+              'Offset': offset, // eslint-disable-line
+              'Content-Length': fileSize,
+              'Content-disposition': 'filename="hello"',
+              'Expect': '', // eslint-disable-line
```
Comment (rssk) [inline src/tusNodeClient.js]: These disables should be removed as well and the props should not be quoted: http://eslint.org/docs/rules/quote-props

File: src/tusNodeClient.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            },
+          };
+
+          const rs = fs.createReadStream(uploadFilePath, {
```
Comment (rssk) [inline src/tusNodeClient.js]: rs = response? Maybe a better name?

File: src/tusNodeClient.js (hunk 3 above)
Comment (rssk) [inline src/tusNodeClient.js]: Just saw this is not your comment, as well as the comment from me below about `rs`. Maybe if you can, clean this up a bit, as you're probably a bit more familiar w/ it than me? Not totally necessary, but it'd be nice to have it cleaner/make more sense, even if it is outside code.

File: src/transferImaging.js (hunk 1 above)
Comment (ruwang) [inline src/transferImaging.js]: This should transfer all files, compared to the meta data. I will rename to transferFiles

File: src/transferImaging.js (hunk 2 above)
Comment (ruwang) [inline src/transferImaging.js]: Currently, no. I know that's resumable in browser. i need to look more to see how the resumable works in nodejs, if that exists in nodejs

Comment (ruwang) [inline src/transferImaging.js]: as a to do.

File: src/tusNodeClient.js (hunk 4 above)
Comment (ruwang) [inline src/tusNodeClient.js]: eslint complains if there is only one return statement inside the block.

File: src/tusNodeClient.js (hunk 3 above)
Comment (ruwang) [inline src/tusNodeClient.js]: sounds good. will do.

File: src/tusNodeClient.js (hunk 6 above)
Comment (ruwang) [inline src/tusNodeClient.js]: sure, readStream. :)

File: src/tusNodeClient.js (hunk 5 above)
Comment (ruwang) [inline src/tusNodeClient.js]: eslint will complain about the hypen `-` if not quoted, thus I quoted `offset` and `expect` as well for consistency, but I think I should leave those two unquoted and remove the eslint comment.

File: src/transferImaging.js (hunk 2 above)
Comment (rssk) [inline src/transferImaging.js]: 👍

File: src/tusNodeClient.js (hunk 4 above)
Comment (rssk) [inline src/tusNodeClient.js]: Right, and as such this line needs to be `() => offset < fileSize; `. If you're using eslint in atom, it'll give you a little dialog that you can click on and it'll take you to the rule. Super helpful to see why you're violating and what are solutions.

Here is the rule: https://github.com/airbnb/javascript#arrows--implicit-return

File: src/tusNodeClient.js (hunk 5 above)
Comment (rssk) [inline src/tusNodeClient.js]: Sounds good 👍

File: src/tusNodeClient.js (hunk 4 above)
Comment (ruwang) [inline src/tusNodeClient.js]: did not realize that. Thanks!

**Pull request merged**